inputgraph/loader.go

```diff
@@ -51,13 +47,26 @@ func newLoader(ctx context.Context, opt HashOpt) *loader {
 		hasher:          hasher.New(),
 		isBaseTarget:    opt.Target.Target == "base",
 		ci:              opt.CI,
-		push:            opt.Push,
 		builtinArgs:     opt.BuiltinArgs,
 		overridingVars:  opt.OverridingVars,
 		earthlyCIRunner: opt.EarthlyCIRunner,
 	}
 }
 
+func (l *loader) handleRun(ctx context.Context, cmd spec.Command) error {
+	opts := commandflag.RunOpts{}
+	_, err := flagutil.ParseArgsCleaned(command.From, &opts, flagutil.GetArgsCopy(cmd))
+	if err != nil {
+		return err
+	}
+
+	if opts.NoCache {
+		return errors.New("--no-cache not supported")
+	}
```

Comment:
I believe that the behavior we want in this case is to skip executing, even if `--no-cache` is used in `RUN`. Same with `RUN --push`. This is because auto-skip is useful to skip deployment if there were no changes to the app being deployed.

This behavior could also back-fire though if you attempt to do a rollback. So we could also just start with saying it's not supported and change the behavior later once we've figured out the usability.

Reply:
Ok, so is the idea that `--auto-skip` overrides the other settings? 

What about when they're specified on the command line? Is a warning ok then?

Reply:
On the command line, I wonder if it should be an error even because they contradict each other.

Reply:
Right, the contradiction is what I'm referring to.